packages/myst-cli/src/transforms/code.ts

```diff
@@ -83,3 +83,48 @@ export function liftCodeMetadataToBlock(session: ISession, filename: string, mda
     }
   });
 }
+
+/**
+ * Traverse mdast, propagate block tags to code and output
+ */
+export function propagateBlockDataToCode(session: ISession, filename: string, mdast: Root) {
+  const blocks = selectAll('block', mdast) as GenericNode[];
```

Comment:
I think it is possible that this might grab too many blocks (e.g. the markdown ones, which aren't going to have the code/output). I think we put a kind or something on the block, not sure where that is at the moment!

Reply:
If this is changed, it's reasonable to change the `liftCodeMetadataToBlock` with the same way. This worth a separate pr to do since two tests will be added and their functionality do not have direct link to the meta tag implementation.

Reply:
That sounds good -- let me know when you are a bit further on the feature and I can go in and test against a few other projects!